sensorimotor/sensorimotor/temporal_pooler.py

```diff
@@ -19,43 +19,42 @@
 # http://numenta.org/licenses/
 # ----------------------------------------------------------------------
 
-import copy
 import numpy
 
 from nupic.bindings.math import (SM32 as SparseMatrix,
-                                SM_01_32_32 as SparseBinaryMatrix,
-                                GetNTAReal,
-                                Random as NupicRandom)
+                                 SM_01_32_32 as SparseBinaryMatrix)
 from nupic.research.spatial_pooler import SpatialPooler
 
-# realDType = GetNTAReal()
 realDType = numpy.float32
 uintType = "uint32"
 
 
-def formatRow(x, formatString = "%d", rowSize = 100):
-  """
-  Utility routine for pretty printing large vectors
-  """
-  s = ''
-  for c,v in enumerate(x):
-    if c > 0 and c % 10 == 0:
-      s += ' '
-    if c > 0 and c % rowSize == 0:
-      s += '\n'
-    s += formatString % v
-  s += ' '
-  return s
-
-
 class TemporalPooler(SpatialPooler):
   """
-  This class implements the new temporal pooler. It tries to form stable and
-  unique representations for sequences that are correctly predicted. If the
-  input is not predicted, it used similar competition rule as the spatial
-  pooler to select active cells
+  This is the default implementation of the new temporal pooler. It attempts
+  to form stable and unique representations of input sequences of active
+  cells from Temporal Memory, but only if those cells were correctly
+  predicted by Temporal Memory. If the active cell sequence was not predicted,
+  a competition, similar to that of the spatial pooler's, is used to select
+  active columns.
+
+  If a column is sufficiently activated by inputs that were predicted in the
+  Temporal Memory, it enters a "pooling mode." A pooling
+  column can, for a limited period, maintain active column status even without
+  receiving any bottom-up input activity. This is implemented using a kind of
+  timer for each column. When the timer "runs out," it loses its pooling status.
+  Whenever significant predicted input returns to a column's synapses,
+  its timer is reset. Note, that in this temporal pooler there is only one cell
+  per column and there is no temporal memory.
   """
 
+  # TODO: RM: Why does this subclass of SpatialPooler inherit so little from
```

Comment:
I agree. It seems kinda weird to me that Temporal Pooler extends SpatialPooler. I'd suggest creating (abstract) common class `Pooler` (or whatever name) that all SP/TP/TM would take as a parent. The common method names (API) and init() and setters/getters could be covered in such a class.